apps/explorer/src/app/components/txs/details/tx-lp-amend.tsx

```diff
@@ -0,0 +1,45 @@
+import React from 'react';
+import { t } from '@vegaprotocol/react-helpers';
+import type {
+  AmendLiquidityProvisionOrder,
+  BlockExplorerTransactionResult,
+} from '../../../routes/types/block-explorer-response';
+import { MarketLink } from '../../links/';
+import type { TendermintBlocksResponse } from '../../../routes/blocks/tendermint-blocks-response';
+import { TxDetailsShared } from './shared/tx-details-shared';
+import { TableCell, TableRow, TableWithTbody } from '../../table';
+
+interface TxDetailsOrderProps {
+  txData: BlockExplorerTransactionResult | undefined;
+  pubKey: string | undefined;
+  blockData: TendermintBlocksResponse | undefined;
+}
+
+/**
+ * Specifies changes to the shape of a users Liquidity Commitment order for
+ * a specific market. So far this only displays the market, which is only
+ * because it's very easy to do so.
+ */
+export const TxDetailsLPAmend = ({
+  txData,
+  pubKey,
+  blockData,
+}: TxDetailsOrderProps) => {
+  if (!txData) {
+    return <>{t('Awaiting Block Explorer transaction details')}</>;
+  }
+
+  const cmd = txData.command as AmendLiquidityProvisionOrder;
```

Comment:
question (not sure): here also, why do the types break without the casting i wonder?

Reply:
actually found the root, resolving this, commenting below instead.

Reply:
Will resolve in https://github.com/vegaprotocol/frontend-monorepo/issues/2176 - but the comment below telling me how to do it better is great regardless